Sources/TokamakDOM/Views/Canvas/Canvas.swift

```diff
@@ -0,0 +1,514 @@
+           .underline: break // Not supported in <canvas>.
+      }
+    }
+
+    canvas
+      .font =
+      .string(
+        "\(style) \(variant) \(weight) \(size ?? "17")pt \(lineHeight) \(family ?? Font.Design.default.families.joined(separator: " "))"
+      )
+  }
+}
+
+extension Canvas: DOMPrimitive {
+  public var renderedBody: AnyView {
+    AnyView(_Canvas(parent: self))
+  }
+}
+
+private extension GraphicsContext.Shading {
+  func cssValue(in environment: EnvironmentValues, with canvas: JSObject,
+                bounds: CGRect) -> JSValue
+  {
+    if case let .resolved(resolved) = _resolve(in: environment)._storage {
+      return resolved.cssValue(in: environment, with: canvas, bounds: bounds)
+    }
+    return .string("none")
+  }
+}
+
+private extension GraphicsContext._ResolvedShading {
+  func cssValue(in environment: EnvironmentValues, with canvas: JSObject,
+                bounds: CGRect) -> JSValue
+  {
+    switch self {
+    case let .levels(palette):
+      guard let primary = palette.first else { break }
+      return primary.cssValue(in: environment, with: canvas, bounds: bounds)
+    case let .style(style):
+      return style.cssValue(in: environment, with: canvas, bounds: bounds)
+    case let .gradient(gradient, geometry, _):
+      let gradientBounds = CGRect(origin: .zero, size: .init(width: 1, height: 1))
+      switch geometry {
+      case let .axial(startPoint, endPoint):
+        return _ResolvedStyle.gradient(
+          gradient,
+          style: .linear(
+            startPoint: .init(x: startPoint.x, y: startPoint.y),
+            endPoint: .init(x: endPoint.x, y: endPoint.y)
+          )
+        ).cssValue(in: environment, with: canvas, bounds: gradientBounds)
+      case let .conic(center, angle):
+        return _ResolvedStyle.gradient(
+          gradient,
+          style: .angular(
+            center: .init(x: center.x, y: center.y),
+            startAngle: .degrees(0),
+            endAngle: angle
+          )
+        ).cssValue(in: environment, with: canvas, bounds: gradientBounds)
+      case let .radial(center, startRadius, endRadius):
+        return _ResolvedStyle.gradient(
+          gradient,
+          style: .radial(
+            center: .init(x: center.x, y: center.y),
+            startRadius: startRadius,
+            endRadius: endRadius
+          )
+        ).cssValue(in: environment, with: canvas, bounds: gradientBounds)
+      }
+    case .tiledImage:
+      break
+    }
+    return .string("none")
+  }
+}
+
+private extension _ResolvedStyle {
+  func cssValue(
+    in environment: EnvironmentValues,
+    opacity: Float = 1,
+    with canvas: JSObject,
+    bounds: CGRect
+  ) -> JSValue {
+    switch self {
+    case let .color(color):
+      return .string(AnyColorBox.ResolvedValue(
+        red: color.red,
+        green: color.green,
+        blue: color.blue,
+        opacity: color.opacity * Double(opacity),
+        space: color.space
+      ).cssValue)
```

Comment:
Maybe this could moved into some extension on `AnyColorBox.ResolvedValue` that has an initializer taking a `color` value and supplied `opacity`. That would reduce these 6 lines to 1 with just that helper initializer call, also potentially resolving the linter warning.